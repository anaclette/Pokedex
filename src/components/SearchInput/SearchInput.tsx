import React, {useState} from 'react';
import {Platform, TextInput, View} from 'react-native';
import {useSafeAreaInsets} from 'react-native-safe-area-context';
import {useDebouncer} from '../../hooks/useDebouncer';
import {useEffect} from 'react';
import {styles} from './searchInput.style';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';

interface Props {
  onDebounce: (userInput: string) => void;
}

export const SearchInput = ({onDebounce}: Props) => {
  const [inputValue, setInputValue] = useState('');
  const debouncedValue = useDebouncer(inputValue, 1500);
  const {top} = useSafeAreaInsets();
  const isIos = Platform.OS === 'ios';
  const topValue = isIos ? top : top + 10;

  useEffect(() => {
    onDebounce(debouncedValue);
  }, [debouncedValue, onDebounce]);

  return (
    <View style={{top: topValue, ...styles.container}}>
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        value={inputValue}
        onChangeText={setInputValue}
        style={{...styles.textInput, height: top}}
        placeholder="Search"
        placeholderTextColor={'gray'}
      />
      <Icon style={styles.searchIcon} name="magnify" size={30} color="brown" />
    </View>
  );
};
