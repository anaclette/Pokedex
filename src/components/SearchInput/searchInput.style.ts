import {StyleSheet} from 'react-native';
import colors from '../../themes/colors';
import fonts from '../../themes/fonts';
import metrics from '../../themes/metrics';

export const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    position: 'absolute',
    width: '90%',
    alignSelf: 'center',
    zIndex: 2,
    minHeight: metrics.scale(38),
  },
  textInput: {
    color: colors.burgundy,
    backgroundColor: colors.translucidLightBackground,
    height: '100%',
    borderColor: colors.gray,
    borderWidth: StyleSheet.hairlineWidth,
    borderRadius: metrics.scale(20),
    paddingLeft: '3%',
    fontFamily: fonts.secondaryText,
  },
  searchIconButton: {
    position: 'absolute',
    alignSelf: 'flex-end',
    justifyContent: 'center',
    right: '4%',
  },
});
