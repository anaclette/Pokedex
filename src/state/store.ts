import {combineReducers, configureStore} from '@reduxjs/toolkit';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {persistReducer, persistStore} from 'redux-persist';
import authReducer from './reducers/authReducer';
import pokemonsReducer from './reducers/pokemonsReducer';
import favouritesReducer from './reducers/favouritesReducer';
import {STATE_MODULES} from '../common/constants';
import {i18nReducer} from './reducers/i18nReducer';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
};

const rootReducer = combineReducers({
  [STATE_MODULES.AUTH]: authReducer,
  [STATE_MODULES.POKEMONS]: pokemonsReducer,
  [STATE_MODULES.FAVOURITES]: favouritesReducer,
  [STATE_MODULES.LANGUAGES]: i18nReducer.reducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      immutableCheck: false,
      serializableCheck: false,
    }),
});

const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export {persistor, store};
