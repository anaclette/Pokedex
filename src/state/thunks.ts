import {createAsyncThunk} from '@reduxjs/toolkit';
import {ACTIONS, baseUrl, STATE_MODULES} from '../common/constants';

export const getPokemons = createAsyncThunk(
  `${STATE_MODULES.POKEMONS}/${ACTIONS.GET_POKEMONS}`,
  async (_, {rejectWithValue}) => {
    try {
      const pokemons = await fetch(
        `${baseUrl}/${STATE_MODULES.POKEMONS}?limit=50&offset=50`,
      )
        .then(response => response.json())
        .then(data => data);
      return pokemons;
    } catch (error) {
      rejectWithValue(error);
    }
  },
);
