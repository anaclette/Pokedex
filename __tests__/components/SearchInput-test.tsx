import * as React from 'react';
import {render} from '@testing-library/react-native';
import SearchInput from '../../src/components/SearchInput';
import {Provider} from 'react-redux';
import {store} from '../../src/state/store';

const onDebounceMock = jest.fn();

describe('SearchInput test', () => {
  const component = (
    <Provider store={store}>
      <SearchInput onDebounce={onDebounceMock} />
    </Provider>
  );
  it('renders', () => {
    render(component);

    expect(component).toMatchSnapshot();
  });
});
