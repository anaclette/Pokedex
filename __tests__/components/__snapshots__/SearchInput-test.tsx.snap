// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput test renders 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SearchInput
    onDebounce={
      [MockFunction] {
        "calls": Array [
          Array [
            "",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  />
</Provider>
`;
